src/wasm2js.h

```diff
@@ -338,6 +338,8 @@ Ref Wasm2JSBuilder::processWasm(Module* wasm, Name funcName) {
       }
       runner.addDefaultOptimizationPasses();
       runner.add("avoid-reinterprets");
+    } else {
+      runner.add("optimize-instructions");
```

Comment:
why is this added? if it makes sense, let's do it in a separate PR with a full explanation.

Reply:
Sure. Will remove this